app/Filament/Concerns/SemesterSelectForOperator.php

```diff
@@ -0,0 +1,113 @@
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace App\Filament\Concerns;
+
+use AdvisingApp\StudentDataModel\Models\Enrollment;
+use Filament\Forms\Components\Select;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Model;
+
+trait SemesterSelectForOperator
+{
+    public static function semesterSelect(): Select
+    {
+        return Select::make('semesters')
+            ->label('Semester')
+            ->options(static::getSemesterOptions())
+            ->searchable()
+            ->multiple();
+    }
+
+    /**
+     * @return array<string, string>
+     */
+    public static function getSemesterOptions(): array
+    {
+        return Enrollment::query()
+            ->select('name')
+            ->distinct()
+            ->orderBy('name')
+            ->get()
+            ->mapWithKeys(fn ($enrollment) => [$enrollment->name => $enrollment->name])
+            ->toArray();
+    }
+
+    /**
+     * @param  Builder<Model>  $query
+     *
+     * @return Builder<Model>
+     */
+    public function applyToBaseQuery(Builder $query): Builder
+    {
+        $relationshipName = $this->constraint->getRelationshipName();
+        $count = $this->settings['count'] ?? 1;
+        $semesters = $this->settings['semesters'] ?? null;
+
+        if (is_null($semesters)) {
+            $semesters = [];
+        } elseif (! is_array($semesters)) {
+            $semesters = [$semesters];
+        }
+
+        $semesters = array_values(array_filter($semesters, fn ($semester) => $semester !== null && $semester !== ''));
+        $lowerSemesters = array_map(fn ($semester) => mb_strtolower($semester), $semesters);
+
+        return $query->whereHas($relationshipName, function (Builder $query) use ($lowerSemesters) {
+            if (! empty($lowerSemesters)) {
+                $placeholders = implode(',', array_fill(0, count($lowerSemesters), '?'));
+                $query->whereRaw("LOWER(name) IN ({$placeholders})", $lowerSemesters);
+            }
+        }, '>=', $count);
```

Comment:
I could be wrong, so do feel free to push back if this works.

But I am trying to understand how doing a `>=` count for this works for all the different operators.

For example the the `Has less than` operator seems like it will be broken by this.

Shouldn't the scoping here just not take into account the count or operator and just be scoping to only records with that semester name?

Also like above this needs to be operating on `semester_name` to `name`